substrate/frame/verify-signature/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Transaction extension which validates a signature against a payload constructed from a call and
+//! the rest of the transaction extension pipeline.
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod extension;
+#[cfg(test)]
+mod tests;
+pub mod weights;
+
+extern crate alloc;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub use benchmarking::BenchmarkHelper;
+use codec::{Decode, Encode};
+pub use extension::VerifySignature;
+use frame_support::Parameter;
+pub use weights::WeightInfo;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
```

Comment:
I was initially a bit lost as to why this has to be a pallet. You need access to a few types like `Signature`, but you can also get those from generics on a standalone extension.

But I now see that there is a also the matter of benchmarking. We don't really have the machinery to benchmark extensions outside of a pallet. 

Am I missing any other points?

Reply:
You're right. It's a benchmarking thing, though I personally like moving the generics from the extension definition to the pallet itself, I think it's cleaner.